workflowproj/camelschema.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflowproj
+
+// camelSchema version 3.20.5. We can update this schema as we go, or add support to more than one in the future.
```

Comment:
We could fetch from the internet, but CLIs that can't access external resources would fail to load. It's a fair question. If you have another idea, we could try in another PR, maybe?

Reply:
@ricardozanini 
In case of updates of the remote camel's file we have to change this file, isn't better to have a copy of this file external to the code, or put the content on a template file and load at the startup when is needed ? An external file is simple to edit and save instead with this approach we need to compile and redeploy an entire operator, image if only this file change, and we have to redeploy on the catalog/hub a new version/csv/olm/other configs files files only for a single line change on this. In the future this could be a separate module and the operator at the startup load the latest version without internal changes

Reply:
can't we delegate this validation to the engine or to the quarkus-camel extension(idk if the extension validate is as well.)?

Reply:
> isn't better to have a copy of this file external to the code, or put the content on a template file and load at the startup when is needed 

In this use case, it's not. Usually we don't pack resources/templates in golang projects. Yet, this will be used by a CLI application. I'd say that's way better to align the Camel version upgrade on runtimes and also do it here using CI/automation instead.

Also, this is a corner case where we try to guess the contents of a file. It's not strict to the project, also it will change once I implement https://issues.redhat.com/browse/KOGITO-9263

> image if only this file change, and we have to redeploy on the catalog/hub a new version/csv/olm/other configs files files only for a single line change on this

We will only change this if Kogito changes its Camel version, which require a new build/version of the whole platform. This will be just yet another change.

> can't we delegate this validation to the engine or to the quarkus-camel extension(idk if the extension validate is as well.)?

Using quarkus/java here?